nautobot/extras/views.py

```diff
@@ -970,10 +978,20 @@ def get(self, request, class_path=None, slug=None):
 
     def post(self, request, class_path=None, slug=None):
         job_model = self._get_job_model_or_404(class_path, slug)
+        template_name = "extras/job.html"
+        try:
+            job_class = job_model.job_class()
+            job_form = job_class.as_form(request.POST, request.FILES) if job_model.job_class is not None else None
+            if hasattr(job_class, "template_name"):
+                try:
+                    get_template(job_class.template_name)
+                    template_name = job_class.template_name
+                except TemplateDoesNotExist as err:
+                    messages.error(request, f'Unable to render requested custom job template "{template_name}": {err}')
+        except RuntimeError as err:
+            messages.error(request, f"Unable to run or schedule '{job_model}': {err}")
+            return redirect("extras:job_list")
```

Comment:
Actually, it occurs to me that on the `post` case, we only actually care about the template name in the case where we weren't able to submit the form successfully and fall back to rendering the template again ([line 1052/1070 below](https://github.com/nautobot/nautobot/pull/1866/files#diff-6dd19ea99cb9defb4b43f5c663a2d2e13821f93e1429b6b372452c1ec3eb06c4R1070)). So it might make sense to move this added logic down to there instead - what do you think?